Data-Structures/src/main/java/linkedList/LinkedList.java

```diff
@@ -45,4 +45,100 @@ public String toString() {
 
         return result.toString();
     }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    // adds a new node with the given value to the end of the list
+    public void append(T value) {
+        Node currentNode = head;
+
+        if (head == null) {
+            Node newNode = new Node(value);
+            head = newNode;
+            return;
+        }
+        // Get to the last node of the list
+        while (currentNode.next != null) {
+            currentNode = currentNode.next;
+        }
+
+        Node newNode = new Node(value);
+        currentNode.next = newNode;
+    }
+
+    // add a new node with the given newValue immediately before the first value node.
+    // Return true or false depends on if we found the value or not
+    public boolean insertBefore(T value, T newVal) {
+        Node currentNode = head;
+
+        // if the value is the first thing on the list, make new node the head and point it to the previous head
+        if (currentNode == null) {
+            return false;
+        } else if (currentNode.value == value) {

```

Comment:
Good work realizing you have an edge case here that must be dealt with differently